src/Native/System.Native/pal_mount.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "../config.h"
+#include "pal_mount.h"
+#include <assert.h>
+#include <string.h>
+#include <errno.h>
+
+// Check if we should use getmntinfo or /proc/mounts
+#if HAVE_MNTINFO
+#include <sys/mount.h>
+#else
+#include <sys/statfs.h>
+#include <mntent.h>
+#define STRING_BUFFER_SIZE 8192
+#endif
+
+#if HAVE_MNTINFO
+
+int GetMountInfo(MountPointFound onFound)
+{
+    // getmntinfo returns pointers to OS-internal structs, so we don't need to worry about free'ing the object
+    int result = 0;
+    struct statfs* mounts = nullptr;
+    int count = getmntinfo(&mounts, 0);
+    if (count > 0)
+    {
+        for (int32_t i = 0; i < count; i++)
+        {
+            int32_t nameLength = strlen(mounts[i].f_mntonname);
+            onFound(mounts[i].f_mntonname, nameLength);
+        }
+    }
+    else
+    {
+        // Something went wrong so set our result to errno
+        result = errno;
```

Comment:
I'm all for keeping with the convention of the underlying function, but getmntinfo either returns 0 or -1...it seemed like it would save a step or having to get errno by having the PAL function return 0 or the error code. This makes it immediately obvious (and easy to determine) what the cause of error was without having to make any more function calls in C#. I'm happy to roll this back if you don't think this design is best, however.